src/logical/model.rs

```diff
@@ -371,19 +371,41 @@ impl Filter {
 pub struct Rule {
     /// Head atoms of the rule
     head: Vec<Atom>,
-    /// Body literals of the rule
-    body: Vec<Literal>,
+    /// Positive Body literals of the rule
+    positive_body: Vec<Atom>,
     /// Filters applied to the body
-    filters: Vec<Filter>,
+    positive_filters: Vec<Filter>,
+    /// Negative Body literals of the rule
+    negative_body: Vec<Atom>,
+    /// Filters applied to the body
+    negative_filters: Vec<Filter>,
```

Comment:
This greatly enlarges the class of rules that we will not be able to serialise back into their original syntactic representation, as we lose the relative order of positive and negative literals. Instead, we should really push #86.

Reply:
Yes this is of course temporary until #86 get done.